src/main/java/bio/terra/service/snapshot/flight/create/NotifyUserOfSnapshotCreationStep.java

```diff
@@ -0,0 +1,37 @@
+package bio.terra.service.snapshot.flight.create;
+
+import bio.terra.service.auth.iam.IamService;
+import bio.terra.service.job.DefaultUndoStep;
+import bio.terra.service.snapshotbuilder.SnapshotBuilderService;
+import bio.terra.service.snapshotbuilder.SnapshotRequestDao;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.exception.RetryException;
+import java.util.UUID;
+
+public class NotifyUserOfSnapshotCreationStep extends DefaultUndoStep {
+  private final SnapshotBuilderService snapshotBuilderService;
+  private final SnapshotRequestDao snapshotRequestDao;
+  private final IamService iamService;
+  private final UUID snapshotRequestId;
+
+  public NotifyUserOfSnapshotCreationStep(
+      SnapshotBuilderService snapshotBuilderService,
+      SnapshotRequestDao snapshotRequestDao,
+      IamService iamService,
+      UUID snapshotRequestId) {
+    this.snapshotBuilderService = snapshotBuilderService;
+    this.snapshotRequestDao = snapshotRequestDao;
+    this.iamService = iamService;
+    this.snapshotRequestId = snapshotRequestId;
+  }
+
+  @Override
+  public StepResult doStep(FlightContext flightContext)
```

Comment:
Note that a side effecting operation like this is not idempotent, so it doesn't conform to the stairway step guidelines. It's possible for this step to be re-run and in that case multiple emails will be sent. In practice this should be very rare though, as it can only arise if the step is run and then the flight is somehow interrupted before stairway can persist the step status to the database.

Reply:
if this kind of functionality does not fit within the parameters of what we want for a step in a flight, are we sure we want to include it in the flight? Instead of after the flight is completed, maybe in the service code?

Reply:
Right now I don't see a better way to do this. We don't have a way to add code to run on flight completion other than adding it to a step, which is part of the flight. While this isn't great, I think it will be OK in practice and it seems unlikely to me that extra emails would be sent.